presto-parquet/src/test/java/com/facebook/presto/parquet/writer/TestParquetWriter.java

```
@@ -77,6 +86,66 @@ public void testRowGroupFlushInterleavedColumnWriterFallbacks()
+                TIMESTAMP.writeLong(pageBuilder.getBlockBuilder(4), 100);
+                pageBuilder.declarePosition();
+            }
+            parquetWriter.write(pageBuilder.build());
+            parquetWriter.close();
+            FileParquetDataSource dataSource = new FileParquetDataSource(parquetFile);
+            ParquetMetadata parquetMetadata = MetadataReader.readFooter(
+                    dataSource,
+                    parquetFile.length(),
+                    Optional.empty(),
+                    false).getParquetMetadata();
+            MessageType parquetSchema = parquetMetadata.getFileMetaData().getSchema();
+            List<org.apache.parquet.schema.Type> parquetTypes = parquetSchema.getFields();
+
+            org.apache.parquet.schema.Type bigintType = parquetTypes.get(0);
+            assertEquals(bigintType.asPrimitiveType().getPrimitiveTypeName().name(), "INT64");
+
+            org.apache.parquet.schema.Type integerType = parquetTypes.get(1);
+            assertEquals(integerType.asPrimitiveType().getPrimitiveTypeName().name(), "INT32");
+
+            org.apache.parquet.schema.Type varcharType = parquetTypes.get(2);
+            assertEquals(varcharType.asPrimitiveType().getPrimitiveTypeName().name(), "BINARY");
+
+            org.apache.parquet.schema.Type booleanType = parquetTypes.get(3);
+            assertEquals(booleanType.asPrimitiveType().getPrimitiveTypeName().name(), "BOOLEAN");
+
+            org.apache.parquet.schema.Type timestampType = parquetTypes.get(4);
+            LogicalTypeAnnotation timeAnnotation = timestampType.getLogicalTypeAnnotation();
+            assertEquals(((LogicalTypeAnnotation.TimestampLogicalTypeAnnotation) timeAnnotation).getUnit(), LogicalTypeAnnotation.TimeUnit.MILLIS);
+            assertFalse(((LogicalTypeAnnotation.TimestampLogicalTypeAnnotation) timeAnnotation).isAdjustedToUTC());
+        }
+        catch (Exception e) {
```

Just let it throw (add `throws Exception` to method signature)